### PR TITLE
Apply `prettier` to pre-commit for automatic linting of `.md` `.yaml` files

### PR DESCRIPTION
**Pull request by bobleesj**
@sbillinge 

Do we want to automate linting for `.md` and `.yaml` files with the following addition to `.pre-commit-config.yaml`?

```
  # prettier - multi formatter for json, yaml, md
  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: f12edd9c7be1c20cfa42420fd0e6df71e42b51ea # frozen: v4.0.0-alpha.8
    hooks:
      - id: prettier
        additional_dependencies:
          - "prettier@^3.2.4"
```

No manual edits have been made.

### CODE REVIEW

File: .github/workflows/build-wheel-release-upload.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
   workflow_dispatch:
   push:
     tags:
-      - '*'  # Trigger on all tags initially, but tag and release privilege are verified in _build-wheel-release-upload.yml
+      - "*" # Trigger on all tags initially, but tag and release privilege are verified in _build-wheel-release-upload.yml
```
Comment (bobleesj) [inline .github/workflows/build-wheel-release-upload.yml:7]: double quote

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 2]
```diff
@@ -42,5 +42,5 @@ Please let @sbillinge know that all checks are done and the package is ready for
 
 <!-- Before closing this issue, please complete the following: -->
 
-- [ ]  Run tutorial examples and conduct functional testing using the installation guide in the README. Attach screenshots/results as comments.
-- [ ]  Documentation (README, tutorials, API references, and websites) is deployed without broken links or missing figures.
+- [ ] Run tutorial examples and conduct functional testing using the installation guide in the README. Attach screenshots/results as comments.
```
Comment (bobleesj) [inline .github/ISSUE_TEMPLATE/release_checklist.md:45]: extra space detected

**Comment by sbillinge** [comment]
> @sbillinge
> 
> Do we want to automate linting for `.md` and `.yaml` files with the following addition to `.pre-commit-config.yaml`?
> 
> ```
>   # prettier - multi formatter for json, yaml, md
>   - repo: https://github.com/pre-commit/mirrors-prettier
>     rev: f12edd9c7be1c20cfa42420fd0e6df71e42b51ea # frozen: v4.0.0-alpha.8
>     hooks:
>       - id: prettier
>         additional_dependencies:
>           - "prettier@^3.2.4"
> ```
> 
> No manual edits have been made.

sure, why not?  prettier is better!

**Comment by bobleesj** [comment]
Okay! Will add to our cookiecutter standard

**Pull request merged**